dagshub/data_engine/model/datasource.py

```diff
@@ -524,10 +551,35 @@ def upload_metadata_from_dataframe(self, df: "pandas.DataFrame", path_column: Op
                 DataFrame with metadata
             path_column: Column with the datapoints' paths. Can either be the name of the column, or its index.
                 If not specified, the first column is used.
+            remote: Whether to upload the metadata via remote or local upload. Default is False.
+                Remote upload means that the data engine API calls will be done from remote machine,
+                rather than from local machine which runs this client.
         """
         self.source.get_from_dagshub()
         send_analytics_event("Client_DataEngine_addEnrichmentsWithDataFrame", repo=self.source.repoApi)
-        self._upload_metadata(self._df_to_metadata(df, path_column, multivalue_fields=self._get_multivalue_fields()))
+
+        if remote:
+            self._remote_upload_metadata_from_dataframe(df, path_column)
+        else:
+            metadata = self._df_to_metadata(df, path_column, multivalue_fields=self._get_multivalue_fields())
+            self._upload_metadata(metadata)
+
+    def _remote_upload_metadata_from_dataframe(
+        self, df: "pandas.DataFrame", path_column: Optional[Union[str, int]] = None
+    ):
+        random_file_name = str(uuid.uuid4())
+        file_path = f"{random_file_name}.parquet"
+        df.to_parquet(file_path, index=False)
+        datasource_name = self.source.name
+
+        task_result = None
+        try:
+            task_result = self.source.repoApi.import_metadata_from_file(datasource_name, file_path, path_column)
+        except:  # noqa # todo better exception handling when we'll have feedback from the backend
+            pass
+        finally:
+            os.remove(file_path)
+        return task_result
```

Comment:
Use [tempfile](https://docs.python.org/3/library/tempfile.html#examples) here (TemporaryDirectory in particular), will save you from doing the os.remove, the lib will do it for you.

Also you don't need to have an `except` here, python is fine with just this:
```
try:
    failing_code()
finally:
    cleanup()
```

Cleanup will always run, and the exception will still be thrown

Reply:
Example: https://github.com/DagsHub/client/blob/34a0c7a44db462e9881125f23c1a3ea93ad5b7ad/dagshub/notebook.py#L95